mm/slab.c

```diff
@@ -111,19 +111,20 @@ static void slab_free(meta_slab_t *slab, void *ptr) {
     new_slab = (slab_t *) ptr;
     /* TODO: eventually below should be done in thread-safe manner */
     list_add_tail(&new_slab->list, &slab->slab_head);
+    decrement_slab_allocs(slab);
 }
 
 /*
  * Round up to nearest power of 2
  * If greater than max size or less than min size, return null
  */
-static void *ktf_alloc(unsigned int size) {
-    unsigned int size_power2 = 0, temp = 0, order_index = 0;
+static void *ktf_alloc(size_t size) {
+    size_t size_power2 = 0, temp = 0, order_index = 0;
     meta_slab_t *slab = NULL, *meta_slab = NULL;
     void *alloc = NULL, *free_page = NULL;
     int ret = 0;
 
-    if (size == 0)
+    if (size <= 0)
         return NULL;
 
     if (size < SLAB_SIZE_MIN)
```

Comment:
What happens when size is larger than SLAB_SIZE_MAX?

Reply:
we return NULL in that case.

Reply:
Ok